importsrv/server_test.go

```diff
@@ -1,137 +1,294 @@
+			workers, // we want to test the parallel case
+			1,       // this field is practically meaningless since we override the flush channel
+			[]metricingester.Sink{cms},
+			[]float64{0.5, 0.95},
+			samplers.AggregateMin|samplers.AggregateMax|samplers.AggregateCount,
+			metricingester.OptFlushChan(flushc), // override the flush ticker channel so we control when flush
+		)
+		s := importsrv.New(ing)
+		defer s.Stop()
 
-func (mi *noopChannelMetricIngester) IngestMetrics(ms []*metricpb.Metric) {
-	mi.in <- ms
-}
+		// EXECUTE THE TEST
 
-func BenchmarkImportServerSendMetrics(b *testing.B) {
-	rand.Seed(time.Now().Unix())
-
-	metrics := metrictest.RandomForwardMetrics(10000)
-	for _, inputSize := range []int{10, 100, 1000, 10000} {
-		ingesters := make([]MetricIngester, 100)
-		for i := range ingesters {
-			ingester := newNoopChannelMetricIngester()
-			ingester.start()
-			defer ingester.stop()
-			ingesters[i] = ingester
-		}
-		s := New(ingesters)
-		ctx := context.Background()
-		input := &forwardrpc.MetricList{Metrics: metrics[:inputSize]}
+		ing.Start()
+		defer ing.Stop()
+		_, err := s.SendMetrics(context.Background(), &forwardrpc.MetricList{Metrics: in})
+		require.NoError(t, err)
+		flushc <- time.Now()
 
-		b.Run(fmt.Sprintf("InputSize=%d", inputSize), func(b *testing.B) {
-			for i := 0; i < b.N; i++ {
-				s.SendMetrics(ctx, input)
+		// COLLECT RESULTS
+
+		var results []samplers.InterMetric
+		for i := 0; i < workers; i++ {
+			select {
+			case result := <-rc:
+				results = append(results, result...)
+			case <-time.After(1 * time.Second):
+				t.Fatal("took too long to flush metric results or no metrics to flush!")
 			}
-		})
+		}
+
+		// ASSERT
+
+		assert.ElementsMatch(
+			t,
+			out,
+			samplers.ToTestMetrics(results),
+			"EXPECTED: %+v\nACTUAL: %+v",
+			out,
+			samplers.ToTestMetrics(results),
+		)
 	}
 }
+
+//func TestOptions_WithTraceClient(t *testing.T) {
+//	c, err := trace.NewClient(trace.DefaultVeneurAddress)
+//	if err != nil {
+//		t.Fatalf("failed to initialize a trace client: %v", err)
+//	}
+//
+//	s := New([]MetricIngester{}, WithTraceClient(c))
+//	assert.Equal(t, c, s.opts.traceClient, "WithTraceClient didn't correctly "+
+//		"set the trace client")
+//}
+//
+//func BenchmarkImportServerSendMetrics(b *testing.B) {
+//	rand.Seed(time.Now().Unix())
+//
+//	metrics := metrictest.RandomForwardMetrics(10000)
+//	for _, inputSize := range []int{10, 100, 1000, 10000} {
+//		ingesters := make([]MetricIngester, 100)
+//		for i := range ingesters {
+//			ingester := newNoopChannelMetricIngester()
+//			ingester.start()
+//			defer ingester.stop()
+//			ingesters[i] = ingester
+//		}
+//		s := New(ingesters)
+//		ctx := context.Background()
+//		input := &forwardrpc.MetricList{Metrics: metrics[:inputSize]}
+//
+//		b.Run(fmt.Sprintf("InputSize=%d", inputSize), func(b *testing.B) {
+//			for i := 0; i < b.N; i++ {
+//				s.SendMetrics(ctx, input)
+//			}
+//		})
+//	}
+//}
```

Comment:
What are these for? If they're intended to be useful, please uncomment, otherwise delete (: